spec/index.bs

```diff
@@ -4782,18 +4802,27 @@ command encoder can no longer be used.
                 descriptor:
             </pre>
 
-            **Returns:** {{undefined}}
-
-            Issue the following steps on the [=Device timeline=] of |this|:
-            <div class=device-timeline>
-                1. If any of the following conditions are unsatisfied, generate a validation
-                    error and stop.
-                    <div class=validusage>
-                        - |this| is [=valid=].
-                        - |this|.{{GPUCommandEncoder/[[debug_group_stack]]}}.length is 0.
+            **Returns:** {{GPUCommandBuffer}}
+
+            Issue the following steps on the [=Content timeline=] of |this|:
+            <div class=content-timeline>
+                1. Let |commandBuffer| be a new {{GPUCommandBuffer}}.
+                1. Issue the following steps on the [=Device timeline=] of |this|:
+                    <div class=device-timeline>
+                        1. If any of the following conditions are unsatisfied, generate a validation
+                            error and stop.
+                            <div class=validusage>
+                                - |this| is [=valid=].
+                                - |this|.{{GPUCommandEncoder/[[debug_group_stack]]}}.length is 0.
+                                - |this|.{{GPUCommandEncoder/[[state]]}} is {{encoder state/open}}.
```

Comment:
I'm worried about the inconsistency of where different internal slots are used and validated.
For example, `[[state]]` is perfectly known on content side, and our spec of `getMappedRange` makes it throw if the state has a wrong value. However, in this PR it's validated on the device timeline.

It would be good to be consistent here. If something *is* needed to be known on the content side, can we at least try to specify it in a way that the validation of this thing *only* happens on the content side, consistently?

Reply:
Feels like we should be indicating which timeline an internal slot will be accessed from when we define them? I wonder if a hungarian notation variant is in order? IE: [[ct_state]] to denote that it's a slot that is tracked and accessed on the content timeline?

Reply:
> For example, `[[state]]` is perfectly known on content side, and our spec of `getMappedRange` makes it throw if the state has a wrong value. However, in this PR it's validated on the device timeline.

Buffer state is (sort of<sup>1</sup>) content-timeline, but command encoder state is device-timeline, so I think this is okay.

---

We need to find a way to extricate our different timelines from one another, but I don't know how. I *think* I'd like to have separate objects for the JS timeline (say, GPUBuffer, having no state), the <sup>1</sup>multi-thread-but-client-side (ct?_Buffer, having the mapping state), and the device timeline (dt_Buffer, having the `destroyed` state). And maybe a concept of a handle (`Handle<dt_Buffer>`)? Some objects may be simpler (most have no multi-thread client state). But I'm still not sure if this is too much crowding.